plugins/org.komodo.eclipse.workspace/src/org/komodo/eclipse/workspace/CorePlugin.java

```diff
@@ -0,0 +1,363 @@
+package org.komodo.eclipse.workspace;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jface.resource.ImageDescriptor;
+import org.eclipse.jface.resource.ImageRegistry;
+import org.eclipse.swt.SWTException;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.ui.plugin.AbstractUIPlugin;
+import org.komodo.core.KEngine;
+import org.komodo.relational.RelationalConstants;
+import org.komodo.relational.model.Column;
+import org.komodo.relational.model.Model;
+import org.komodo.relational.model.Parameter;
+import org.komodo.relational.model.Procedure;
+import org.komodo.relational.model.Table;
+import org.komodo.relational.model.Parameter.Direction;
+import org.komodo.relational.vdb.DataRole;
+import org.komodo.relational.vdb.Permission;
+import org.komodo.relational.vdb.Translator;
+import org.komodo.relational.vdb.Vdb;
+import org.komodo.relational.workspace.WorkspaceManager;
+import org.komodo.repository.LocalRepository;
+import org.komodo.repository.LocalRepository.LocalRepositoryId;
+import org.komodo.shell.Messages;
+import org.komodo.shell.Messages.SHELL;
+import org.komodo.spi.KException;
+import org.komodo.spi.constants.StringConstants;
+import org.komodo.spi.repository.KomodoObject;
+import org.komodo.spi.repository.Repository;
+import org.komodo.spi.repository.Repository.UnitOfWork;
+import org.komodo.spi.repository.RepositoryObserver;
+import org.modeshape.sequencer.teiid.lexicon.CoreLexicon.ModelType;
+import org.osgi.framework.BundleContext;
+
+/**
+ * The activator class controls the plug-in life cycle
+ */
+public class CorePlugin extends AbstractUIPlugin implements StringConstants {
+
+	// The plug-in ID
+	public static final String PLUGIN_ID = "org.komodo.eclipse.workspace"; //$NON-NLS-1$
+	
+	private static final String CORE_REPOSITORY_CONFIG = "workspace-core-plugin-repository-in-memory-config.json";
+
+	// The shared instance
+	private static CorePlugin plugin;
+	
+	private static KEngine kEngine;
+	
+	private static WorkspaceManager wsManager;
+	
+	/**
+	 * The constructor
+	 */
+	public CorePlugin() {
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * @see org.eclipse.ui.plugin.AbstractUIPlugin#start(org.osgi.framework.BundleContext)
+	 */
+	public void start(BundleContext context) throws Exception {
+		super.start(context);
+		plugin = this;
+		
+		kEngine = KEngine.getInstance();
+		
+		try {
+			startKEngine();
```

Comment:
How do we know that this is the first plugin to start and thus THE plugin to start the engine?